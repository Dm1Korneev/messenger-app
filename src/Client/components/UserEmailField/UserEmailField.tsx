import { TextField } from 'formik-material-ui';

type Props = Omit<
  React.ComponentPropsWithoutRef<typeof TextField>,
  'margin' | 'label' | 'fullWidth' | 'autoComplete'>

export const UserEmailField = (props: Props) => (
  <TextField
    autoComplete="email"
    fullWidth
    label="Email Address *"
    margin="normal"
    {...props}
  />
);
