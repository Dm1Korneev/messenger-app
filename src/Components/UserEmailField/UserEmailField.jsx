import React from 'react';
import { TextValidator } from 'react-material-ui-form-validator';
import PropTypes from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';

function UserEmailField(props) {
  const { value, onChange } = props;

  return (
    <TextValidator
      margin="normal"
      label="Email Address *"
      fullWidth
      onChange={onChange}
      name="email"
      id="email"
      autoComplete="email"
      color="primary"
      value={value}
      validators={['required', 'isEmail']}
      errorMessages={['this field is required', 'email is not valid']}
    />
  );
}

UserEmailField.defaultProps = {
  value: undefined,
};
UserEmailField.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

const styles = () => ({});

export default withStyles(styles)(UserEmailField);

