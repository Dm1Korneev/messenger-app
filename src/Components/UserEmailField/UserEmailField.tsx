import React from 'react';
import { TextField } from 'formik-material-ui';

type Props = Omit<
  React.ComponentPropsWithoutRef<typeof TextField>,
  'margin' | 'label' | 'fullWidth' | 'autoComplete'>

export const UserEmailField = (props: Props) => (
  <TextField
    margin="normal"
    label="Email Address *"
    fullWidth
    autoComplete="email"
    {...props}
  />
);
