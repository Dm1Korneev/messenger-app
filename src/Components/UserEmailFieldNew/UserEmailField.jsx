import React from 'react';
import { TextField } from 'formik-material-ui';

const UserEmailField = (props) => (
  <TextField
    margin="normal"
    label="Email Address *"
    fullWidth
    autoComplete="email"
    {...props}
  />
);

export default UserEmailField;

