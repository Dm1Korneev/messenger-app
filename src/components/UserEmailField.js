import React from "react";
import { withStyles } from "@material-ui/core/styles";
import { TextValidator } from "react-material-ui-form-validator";

function UserEmailField(props) {
  const { value, onChange } = props;

  return (
    <TextValidator
      margin="normal"
      label="Email Address *"
      fullWidth
      onChange={onChange}
      name="email"
      id="email"
      autoComplete="email"
      color="primary"
      value={value}
      validators={["required", "isEmail"]}
      errorMessages={["this field is required", "email is not valid"]}
    />
  );
}

const styles = theme => ({});

export default withStyles(styles)(UserEmailField);
