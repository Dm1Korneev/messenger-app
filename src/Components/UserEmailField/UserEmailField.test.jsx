import React from 'react';

import UserEmailField from './UserEmailField';

jest.mock('react-material-ui-form-validator', () => ({
  TextValidator: global.mockComponent('TextValidator'),
}));

describe('render UserEmailField component', () => {
  let wrapper;

  const props = {
    value: 'TEST',
    onChange: jest.fn(),
  };

  beforeAll(() => {
    wrapper = global.mount(
      <UserEmailField {...props} />,
    );
  });

  test('component is render', () => {
    expect(wrapper.find('UserEmailField').length).toBe(1);
  });

  test('TextValidator subcomponent is render', () => {
    wrapper.update();
    expect(wrapper.find({ originalcomponent: 'TextValidator' }).length).toBe(1);
  });
});

