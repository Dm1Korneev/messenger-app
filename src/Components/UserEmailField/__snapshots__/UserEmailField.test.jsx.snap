// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render UserEmailField component snapshot test 1`] = `
<DocumentFragment>
  <div
    data-originalcomponent="TextValidator"
    test-prop-autocomplete="\\"email\\""
    test-prop-color="\\"primary\\""
    test-prop-errormessages="[\\"this field is required\\",\\"email is not valid\\"]"
    test-prop-fullwidth="true"
    test-prop-id="\\"email\\""
    test-prop-label="\\"Email Address *\\""
    test-prop-margin="\\"normal\\""
    test-prop-name="\\"email\\""
    test-prop-validators="[\\"required\\",\\"isEmail\\"]"
    test-prop-value="\\"TEST\\""
  />
</DocumentFragment>
`;
